### PR TITLE
client: add wasm for client transport

### PR DESCRIPTION
**Pull request by Icemap**
Please read https://github.com/etcd-io/etcd/blob/main/CONTRIBUTING.md#contribution-flow.

Add for https://github.com/etcd-io/etcd/issues/16908.

### CODE REVIEW

**Review by jmhbnz — COMMENTED** [review]
Hey @Icemap - Thanks for raising this, to ensure your contribution can be accepted please sign your commit so the developer certificate of origin (DCO) check passes, i.e:

```bash
git rebase --signoff HEAD~2
```

**Comment by Icemap** [comment]
Good catch @jmhbnz. I already made the DCO happy.

**Review by jmhbnz — APPROVED** [review]
LGTM - Thanks @Icemap.

Can I clarify, was this the only issue you encountered when building etcd targeting wasm?

**Comment by Icemap** [comment]
Hi, @jmhbnz.
It's an issue that I encountered when I was building another application.
In this case, ETCD is a dependent package.
I assume it's not the only issue if we build ETCD itself for the wasm target.

**Review by serathius — APPROVED** [review]

**Comment by serathius** [comment]
cc @ahrtr

File: client/pkg/transport/sockopt_wasm.go [hunk 1]
```diff
@@ -0,0 +1,30 @@
+// Copyright 2021 The etcd Authors
```
Comment (ahrtr) [inline client/pkg/transport/sockopt_wasm.go:1]: nit
```suggestion
// Copyright 2023 The etcd Authors
```

Comment (ahrtr) [reply client/pkg/transport/sockopt_wasm.go]: resolved

**Review by ahrtr — APPROVED** [review]
lgtm

**Pull request merged**